eva_assembly_ingestion/assembly_ingestion_job.py

```diff
@@ -0,0 +1,380 @@
+            target_assembly=self.target_assembly,
+            rs_report_path=f'{source_assembly}_to_{self.target_assembly}_rs_report.txt'
+        )
+        with open(output_file_path, 'w') as open_file:
+            open_file.write(properties)
+        return output_file_path
+
+    def set_status(self, source_assembly, status, start_time=None, end_time=None):
+        query = f"UPDATE {self.tracking_table} SET remapping_status='{status}' "
+        if start_time:
+            query += f", remapping_start='{start_time.isoformat()}' "
+        if end_time:
+            query += f", remapping_end='{end_time.isoformat()}' "
+        query += (
+            f"WHERE release_version={self.release_version} "
+            f"AND origin_assembly_accession='{source_assembly}' AND taxonomy={self.taxonomy}"
+        )
+        with get_metadata_connection_handle(self.maven_profile, self.private_settings_file) as pg_conn:
+            execute_query(pg_conn, query)
+
+    def set_status_start(self, source_assembly):
+        self.set_status(source_assembly, 'Started', start_time=datetime.now())
+
+    def set_status_end(self, source_assembly):
+        self.set_status(source_assembly, 'Completed', end_time=datetime.now())
+
+    def set_status_failed(self, source_assembly):
+        self.set_status(source_assembly, 'Failed')
+
+    def set_counts(self, source_assembly, source, nb_variant_extracted=None, nb_variant_remapped=None,
+                   nb_variant_ingested=None):
+        set_statements = []
+        query = (
+            f"SELECT * FROM {self.tracking_table} "
+            f"WHERE release_version={self.release_version} AND origin_assembly_accession='{source_assembly}' "
+            f"AND taxonomy='{self.taxonomy}' AND source='{source}'"
+        )
+        with get_metadata_connection_handle(self.maven_profile, self.private_settings_file) as pg_conn:
+            # Check that this row exists
+            results = get_all_results_for_query(pg_conn, query)
+        if results:
+            if nb_variant_extracted is not None:
+                set_statements.append(f"num_ss_extracted = {nb_variant_extracted}")
+            if nb_variant_remapped is not None:
+                set_statements.append(f"num_ss_remapped = {nb_variant_remapped}")
+            if nb_variant_ingested is not None:
+                set_statements.append(f"num_ss_ingested = {nb_variant_ingested}")
+
+        if set_statements:
+            query = (
+                f"UPDATE {self.tracking_table} SET {', '.join(set_statements)} "
+                f"WHERE release_version={self.release_version} AND origin_assembly_accession='{source_assembly}' "
+                f"AND taxonomy='{self.taxonomy}' AND source='{source}'"
+            )
+            with get_metadata_connection_handle(cfg['maven']['environment'], cfg['maven']['settings_file']) as pg_conn:
+                execute_query(pg_conn, query)
+
+    def count_variants_from_logs(self, assembly_directory, source_assembly):
```

Comment:
It would be good to also get the clustering counts here. maybe in a subsequent ticket.

Reply:
Agreed, though we also need to decide exactly what clustering process we want to be running here, and what the counts for each one should be...

Reply:
Actually they would not go here but as a step after the clustering.